experimental/Origami/src/Origami.jl

```diff
@@ -0,0 +1,309 @@
+# Add your new types, functions, and methods here.
+
+module OrigamiHelper
+
+using ..GAP
+
+function __init__()
+  mod_p = "https://ag-weitze-schmithusen.github.io/ModularGroup/PackageInfo.g"
+  ori_p = "https://ag-weitze-schmithusen.github.io/Origami/PackageInfo.g"
+  GAP.Packages.install(mod_p)
+  GAP.Packages.install(ori_p)
+  GAP.Packages.load("Origami")
+end
+
+end
+
+include("exports.jl")
+include("types.jl")
+include("canonical.jl")
+include("deck_group.jl")
+include("action.jl")
+include("special_origami.jl")
+include("generate.jl")
+include("homology_action.jl")
+include("cyclic_torus_covers.jl")
+include("homology.jl")
+include("normal_origami.jl")
+include("systoles.jl")
+
+@doc raw"""
+    origami(h::PermGroupElem, v::PermGroupElem)
+
+This function constructs an Origami object from two given permutations h and v . A test is
+performed to check whether the surface described by the given permutations is connected, i.e.
+whether the group generated by the two permutations acts transitively.
+
+# Examples
+```jldoctest
+julia> h = @perm (1,2)
+(1,2)
+
+julia> v = @perm (2,3)
+(1,2)
+
+julia> o = origami(h,v)
+Origami ((1,2),(2,3), 3)
+```
+"""
+function origami(h::PermGroupElem, v::PermGroupElem)
+  d = max(degree(h), degree(v))
+  G = symmetric_group(d)
+  obj = GAP.Globals.Origami(GapObj(h), GapObj(v))
+  return Origami(obj, G(h), G(v), d)
+end
+
+# ugly workaround
+function from_GAP_origami(o::GapObj)
+  d = GAP.Globals.DegreeOrigami(o)::Int
+  G = symmetric_group(d)
+  h = PermGroupElem(G, GAP.Globals.HorizontalPerm(o))
+  v = PermGroupElem(G, GAP.Globals.VerticalPerm(o))
+  return origami(h, v)
+end
+
+function origami_disconnected(h::PermGroupElem, v::PermGroupElem, d::Integer)
+  return Origami(GAP.Globals.OrigamiNC(GapObj(h), GapObj(v), d), h, v, d)
+end
+
+function Base.:(==)(a::Origami, b::Origami)
+  # TODO rewrite this? for now use Gap equality
+  return (a.h == b.h) && (a.v == b.v)
+end
+
+function Base.hash(o::Origami, h::UInt=0x000000000)
```

Comment:
```suggestion
function Base.hash(o::Origami, h::UInt)
```
this should not have a default value, as julia base already provides a one-argument `hash` method